frontend/src/components/OperationsTableCard/OperationsTableCard.tsx

```diff
@@ -159,6 +163,55 @@ const OperationsTableCard: FunctionComponent<OperationsTableCardProps> = (props)
     }
   };
 
+  const dismissToast = (toastId: string | number) => {
+    toast.dismiss(toastId);
+  };
+
+  const parseCSV = (id: number, fileName: string, toastId: number | string) => {
+    return new Promise((resolve, reject) => {
+      const csvResults: any[] = [];
+      let columns: string[] | undefined;
+      parse(`${api.routes.EXPORT}${id}/`, {
+        step: function (results) {
+          csvResults.push(results.data);
+          if (!columns) {
+            columns = results.meta.fields;
+          }
+        },
+        download: true,
+        header: true,
+        dynamicTyping: true,
+        error: function () {
+          toast.update(toastId, { render: 'Error', type: 'error', isLoading: false });
+          reject();
+        },
+        complete: function () {
+          const csvExporter = new ExportToCsv({
+            filename: fileName,
+            fieldSeparator: ',',
+            quoteStrings: '"',
+            decimalSeparator: '.',
+            showLabels: true,
+            useTextFile: false,
+            useBom: true,
+            headers: columns && columns.length ? columns : [],
+          });
+
+          toast.update(toastId, { render: 'Success', type: 'success', isLoading: false });
```

Comment:
If possible, account for multiple file being downloaded, so it'd be nice if the success message indicated what was successfully downloaded.

Reply:
Updated